### PR TITLE
TST: splitlines in rec2txt test

### PR DESCRIPTION
**Pull request by tacaswell**
On windows the output has '\r\n' instead of '\n' for new
lines.

If this passes I plan to self-merge to un-break appveyor

### CODE REVIEW

**Comment by tacaswell** [comment]
attn @dopplershift I think this is an equivalent test .

**Comment by tacaswell** [comment]
@janschulz Can you help out with the windows issue here?  I do not have a system to test on and debugging via appveyor seems maddeningly painful!

**Comment by dopplershift** [comment]
👍 from me.

**Comment by tacaswell** [comment]
In windows this seems to output one less space is some cases:

```python
a = ['       x   y   s    s2', '   1.000   2   foo  bing ']
b = ['       x   y   s   s2',  '   1.000   2   foo bing ']
```

**Comment by tacaswell** [comment]
Sorry, had order flipped, windows _adds_ an extra space.

**Comment by tacaswell** [comment]
@dopplershift I think there was a bug in computing the width of the string columns which for some reason was behaving differently on windows/linux.

**Comment by janschulz** [comment]
I have a look... Looks liek there are at least two problems:

The one where the link step [can't find png.lib](https://ci.appveyor.com/project/mdboom/matplotlib/build/1.0.1517/job/uxehvnwi657ks0e0) and the one where there is a [space difference](https://ci.appveyor.com/project/mdboom/matplotlib/build/1.0.1517/job/wph4btf8fkf053ln).

Re the png one: it seems that conda-forge recently added a libpng (and that one is installed), could be that there is a difference to the official one? https://github.com/conda-forge/libpng-feedstock/commits/master (cc: @ocefpaf)

**Comment by tacaswell** [comment]
I think I have the space issue fixed

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -3188,9 +3188,7 @@ def get_justify(colname, column, precision):
         ntype = column.dtype
 
         if np.issubdtype(ntype, str) or np.issubdtype(ntype, bytes):
-            # The division below handles unicode stored in array, which could
-            # have 4 bytes per char
-            length = max(len(colname), column.itemsize // column[0].itemsize)
+            length = max(len(colname), len(column[0]))
```
Comment (dopplershift) [inline lib/matplotlib/mlab.py]: I'm not convinced `len()` will do what we want if, e.g. the data for a column is `['foo', 'bars']`. Probably should change the test to have some differing string sizes and see what happens.

Comment (tacaswell) [inline lib/matplotlib/mlab.py]: is that allowed in recarrays?  The types look like fixed-width string, not variable width.

Comment (dopplershift) [inline lib/matplotlib/mlab.py]:
```
In [18]: a = np.array([(1, 'foo'), (2, 'bars')], dtype=np.dtype([('i', np.int32), ('s', np.str, 4)]))

In [19]: a
Out[19]:
array([(1, 'foo'), (2, 'bars')],
      dtype=[('i', '<i4'), ('s', '<U4')])

In [20]: len(a['s'][0])
Out[20]: 3
```

Comment (dopplershift) [inline lib/matplotlib/mlab.py]: But it turns out my original code was broken as well:
```
In [29]: a['s'][0].itemsize
Out[29]: 3
```
WTF?

Comment (QuLogic) [inline lib/matplotlib/mlab.py]: Maybe you want `a.dtype['s'].itemsize` == 16 ( / 4 for Unicode == 4)?

Comment (tacaswell) [inline lib/matplotlib/mlab.py]: `len()` is not the right thing.  The best I can come up with is `int(a.dtype['s'].str[2:])`

**Comment by tacaswell** [comment]
And tests are passing, but it is dying as while trying to build the conda package

**Comment by tacaswell** [comment]
ok, this is now passing (except for a down-load error on one of the appveyor tests).

I had to turn off the conda package building which is less than great, but makes the test useful again for code review.

**Comment by cgohlke** [comment]
Is this going to be backported? matplotlib 1.5.2rc1 on Windows fails one test:
```
======================================================================
FAIL: test_csv2txt_basic (matplotlib.tests.test_mlab.rec2txt_testcase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "X:\Python35\lib\site-packages\matplotlib\tests\test_mlab.py", line 409, in test_csv2txt_basic
    assert_equal(mlab.rec2txt(a), truth)
AssertionError: '       x   y   s    s2\r\n   1.000   2   foo  bing \r\n   2.[16 chars]lah ' != '       x   y   s   s2\n   1.000   2   foo bing \n   2.000   3   bar blah '
-        x   y   s    s2
?                 -     -
+        x   y   s   s2
-    1.000   2   foo  bing
?                   -      -
+    1.000   2   foo bing
-    2.000   3   bar  blah ?                   -
+    2.000   3   bar blah

----------------------------------------------------------------------
Ran 5209 tests in 1399.371s
```

**Comment by tacaswell** [comment]
Yes, sorry this got lost in the appveyor related issues.

**Comment by tacaswell** [comment]
backported to v1.5.x as cab8cf7

**Pull request merged**